Gives artificers feedback when healing other constructs plus a few sprite fixes

ChangelingRain opened the pull request:
:cl: Joan
rscadd: Artificers now have feedback when healing other constructs, showing how much health the target has.
tweak: Constructs can move in space.
imageadd: Fixes cult flooring not lining up with normal floors.
imageadd: Fixes a juggernaut back spine not glowing where it should.
/:cl:

ChangelingRain committed:
Gives artificers feedback when healing another construct, showing how much health it has.
Constructs can into space.
Fixes cult flooring not lining up with normal floors, I don't care if you're evil, this is just wrong.
Fixes a juggernaut back spine being not glowy enough.

ChangelingRain committed:
YOU CAN'T REPAIR DENTS THAT DON'T EXIST

Anonus commented on code/modules/mob/living/simple_animal/constructs.dm:
```diff
@@ -54,9 +56,19 @@
+			if(src != M)
+				M << "<span class='cult'>You cannot repair <EM>[src]'s</EM> dents, as it has none!</span>"
+			else
+				M << "<span class='cult'>You cannot repair your own dents, as you have none!</span>"
```
This entire thing is pretty bad is there a way I can make it simpler somehow

KorPhaeron commented:
>constructs can move in space.

They already could though git gud

phil235 commented on code/modules/mob/living/simple_animal/constructs.dm:
```diff
@@ -14,6 +14,8 @@
 	status_flags = CANPUSH
 	attack_sound = 'sound/weapons/punch1.ogg'
+	environment_smash = 1
+	see_in_dark = 7
```
undocumented features?

phil235 commented on code/modules/mob/living/simple_animal/constructs.dm:
```diff
@@ -54,9 +56,19 @@
 /mob/living/simple_animal/construct/attack_animal(mob/living/simple_animal/M)
 	if(istype(M, /mob/living/simple_animal/construct/builder))
 		adjustBruteLoss(-5)
```
you could move `adjustBruteLoss(-5)` after `if(health < maxHealth)`

ChangelingRain committed:
Removes terrible, unnecessary bullet_act() from constructs, replacing it with the proper damage_coeff list.
I'd bet money it was there just so ebows couldn't damage constructs.

ChangelingRain committed:
oldcode deserves to die
juggernaut reflection uses the apply damage proc correctly instead of checking if the damage type is brute or burn

ChangelingRain committed:
Conflixed
